riscv/insns/ssamoswap_d.h

```diff
@@ -0,0 +1,20 @@
+require_extension(EXT_ZICFISS);
+require_extension('A');
+require_rv64;
+reg_t s = STATE.mstatus->read();
+reg_t prev_prv = get_field(s, MSTATUS_MPP);
+reg_t mSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
+
+if (prev_prv != PRV_M && !mSSE)
+  throw trap_illegal_instruction(insn.bits());
+
+if (!p->extension_enabled('S'))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_U && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_S && STATE.v && !get_field(STATE.henvcfg->read(), HENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_U && STATE.v && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else
+  WRITE_RD(MMU.amo<uint64_t>(RS1, [&](uint64_t UNUSED lhs) { return RS2; }));
```

Comment:
How does this code indicate to MMU that the amo should be performed using shadow stack load and stores?

Reply:
it may suffice to be just <T> since there is only one amo operation - so we could call this `ssamoswap`